### PR TITLE
subsys: power: remove device_pm_control_nop function

### PR DESCRIPTION
**Pull request by gmarull**
Devices that do not require PM should just use `NULL`. `device_pm_control_nop` is still kept as an alias to `NULL` untill all
in-tree usage is replaced with `NULL`. After that it can be either removed or deprecated to make out-of-tree users transition smoother.

Some devices have been ported (everything except `drivers/*`, the main user of `device_pm_control_nop`). I'll take care of porting remaining devices if this PR moves forward.

### CODE REVIEW

File: include/device.h [hunk 1]
```diff
@@ -781,15 +762,12 @@ static inline int device_set_power_state(const struct device *dev,
...
-	if (dev->device_pm_control) {
-		return dev->device_pm_control(dev,
-						 DEVICE_PM_SET_POWER_STATE,
-						 &device_power_state, cb, arg);
-	} else {
-		return device_pm_control_nop(dev,
-						 DEVICE_PM_SET_POWER_STATE,
-						 &device_power_state, cb, arg);
+	if (dev->device_pm_control ==  NULL) {
+		return -ENOTSUP;
```
Comment (nashif) [inline include/device.h:766]: should return -ENOSYS (not implemented)

File: include/device.h [hunk 2]
```diff
@@ -808,15 +786,12 @@ static inline int device_set_power_state(const struct device *dev,
...
-	if (dev->device_pm_control) {
-		return dev->device_pm_control(dev,
-						 DEVICE_PM_GET_POWER_STATE,
-						 device_power_state, NULL, NULL);
-	} else {
-		return device_pm_control_nop(dev,
-						 DEVICE_PM_GET_POWER_STATE,
-						 device_power_state, NULL, NULL);
+	if (dev->device_pm_control == NULL) {
+		return -ENOTSUP;
```
Comment (nashif) [inline include/device.h:790]: -ENOSYS

File: include/device.h (hunk 1 above)
Comment (gmarull) [reply include/device.h]: fixed, thanks (note: `-ENOTSUP` was returned from `device_pm_control_nop`).

File: include/device.h (hunk 2 above)
Comment (gmarull) [reply include/device.h]: fixed

**Review by erwango — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**